src/rgw/rgw_sal_motr.cc

```diff
@@ -2690,12 +2689,6 @@ int MotrMultipartUpload::list_parts(const DoutPrefixProvider *dpp, CephContext *
                                               << "], meta:pvid=[" << meta.pver.f_container << "," << meta.pver.f_key
                                               << "], meta:layout id=" << meta.layout_id << dendl;
 
-    if (!expected_next)
-      expected_next = info.num + 1;
-    else if (expected_next && info.num != expected_next)
```

Comment:
if we want to keep `expected_next` variable for some reason, we can change `!=` to `<`.
```c++
else if (expected_next && info.num < expected_next)
```

Reply:
@sumedhak27 : I'm ok if you remove the entire 'expected_next' logic. However, make sure Complete API fails on such uploaded parts (Try to run complete API on this data and see). Also, check the behavior of Ceph with Rados backend and AWS S3 for such uploaded parts.

Reply:
<details><summary>Multipart upload Case</summary>

```bash
╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api create-multipart-upload --bucket mybkt --key mpobj
{
    "Bucket": "mybkt",
    "Key": "mpobj",
    "UploadId": "2~iJnSagjxER7T8R5jAG34_cIjikLqsdt"
}

╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api upload-part --bucket mybkt --key mpobj --upload-id 2~iJnSagjxER7T8R5jAG34_cIjikLqsdt --part-number 10000 --body ~/testfiles/1mb.txt
{
    "ETag": "\"9cce32b2829b5c73b70075cef7210458\""
}

╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api upload-part --bucket mybkt --key mpobj --upload-id 2~iJnSagjxER7T8R5jAG34_cIjikLqsdt --part-number 1 --body ~/testfiles/10mb.file
{
    "ETag": "\"311175294563b07db7ea80dee2e5b3c6\""
}

╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api list-parts --bucket mybkt --key mpobj --upload-id 2~iJnSagjxER7T8R5jAG34_cIjikLqsdt
{
    "Parts": [
        {
            "PartNumber": 1,
            "LastModified": "2022-03-17T07:19:58.729Z",
            "ETag": "\"311175294563b07db7ea80dee2e5b3c6\"",
            "Size": 10000000
        },
        {
            "PartNumber": 10000,
            "LastModified": "2022-03-17T07:19:43.764Z",
            "ETag": "\"9cce32b2829b5c73b70075cef7210458\"",
            "Size": 1048576
        }
    ],
    "Initiator": null,
    "Owner": {
        "DisplayName": "",
        "ID": ""
    },
    "StorageClass": "STANDARD"
}

╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  vim ~/testfiles/mpobj.json

╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api complete-multipart-upload --bucket mybkt --key mpobj --upload-id 2~iJnSagjxER7T8R5jAG34_cIjikLqsdt  --multipart-upload=file:///root/testfiles/mpobj.json
{
    "Location": "http://localhost:8000/mybkt/mpobj",
    "Bucket": "mybkt",
    "Key": "mpobj",
    "ETag": ""
}

╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api head-object  --bucket mybkt --key mpobj
{
    "AcceptRanges": "bytes",
    "LastModified": "Thu, 17 Mar 2022 07:26:58 GMT",
    "ContentLength": 11048576,
    "ETag": "\"c09f50236e31d05bc7a403ab77e1f69c-2\"",
    "ContentType": "binary/octet-stream",
    "Metadata": {}
}
```

</details>

Reply:
<details><summary>Abort multipart upload case</summary>

```bash
╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api create-multipart-upload --bucket mybkt --key mpobj1
{
    "Bucket": "mybkt",
    "Key": "mpobj1",
    "UploadId": "2~oiJz5zx1ckcuafCOEm1x9Q8gSIruch5"
}
╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api upload-part --bucket mybkt --key mpobj1 --upload-id 2~oiJz5zx1ckcuafCOEm1x9Q8gSIruch5 --part-number 10000 --body ~/testfiles/1mb.txt
{
    "ETag": "\"9cce32b2829b5c73b70075cef7210458\""
}
╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api list-multipart-uploads --bucket mybkt
{
    "Uploads": [
        {
            "UploadId": "2~oiJz5zx1ckcuafCOEm1x9Q8gSIruch5",
            "Key": "mpobj1",
            "Initiated": "2022-03-17T07:32:00.754Z",
            "StorageClass": "STANDARD",
            "Owner": {
                "DisplayName": "",
                "ID": ""
            },
            "Initiator": {
                "ID": "",
                "DisplayName": ""
            }
        }
    ]
}
╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api abort-multipart-upload --bucket mybkt --key mpobj1 --upload-id 2~oiJz5zx1ckcuafCOEm1x9Q8gSIruch5
╭─root@ssc-vm-g4-rhev4-0318.colo.seagate.com ~/cortx/cortx-rgw/build  ‹CORTX-29411-FixListNonConsecutiveParts›
╰─➤  aws s3api list-multipart-uploads --bucket mybkt
```

</details>